### PR TITLE
fix: fix deprecated subdependencies found warn(#248)

### PR DESCRIPTION
**Pull request by 0x0bit**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by mcollina** [comment]
Those are only devDependencies, they won't affect the actual warnings on install

**Comment by MrYbo** [comment]
> Those are only devDependencies, they won't affect the actual warnings on install

Development dependencies are often installed and used during development, and by default, the npm install command installs all dependencies in the package.json file, including dependencies and devDependencies.

**Comment by mcollina** [comment]
@gurgunday ptal.

I think this should be rebased on the `next` branch.

**Comment by MrYbo** [comment]
What is the current status of this？

**Comment by gurgunday** [comment]
Taking a look

File: .taprc [hunk 1]
```diff
@@ -1,6 +1,7 @@
...
-flow: false
-coverage: true
-check-coverage: true
+# vim: set filetype=yaml :
+node-arg:
+  - '--allow-natives-syntax'
+
+include:
+  - 'test/*.test.js'
+  - 'test/*.test.ts'
```
Comment (Uzlopak) [inline .taprc:7]: we dont have ts files

```suggestion
```

File: .taprc [hunk 2]
```diff
@@ -1,6 +1,7 @@
-100: true
-ts: false
-jsx: false
-flow: false
-coverage: true
-check-coverage: true
+# vim: set filetype=yaml :
+node-arg:
+  - '--allow-natives-syntax'
```
Comment (Uzlopak) [inline .taprc:3]: why natives syntax?

**Comment by Uzlopak** [comment]
Missing change in readme.md regarding code style button

File: eslint.config.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+'use strict'
+
+const neo = require('neostandard')
+
+module.exports = [
+  ...neo(),
+  {
+    rules: {
+      '@stylistic/comma-dangle': ['error', {
```
Comment (gurgunday) [inline eslint.config.js:9]: I think this was changed upstream

**Comment by climba03003** [comment]
> by default, the npm install command installs all dependencies in the package.json file, including dependencies and devDependencies.

I don't think that is the truth.

```shell
mkdir repro
cd repro
npm install @fastify/helmet
```

You can see only `dependencies` is installed.

**Comment by gurgunday** [comment]
Interesting, normally it should also install devDeps

https://docs.npmjs.com/cli/v10/commands/npm-install#description

Maybe you have NODE_ENV set to production?

**Comment by climba03003** [comment]
> Interesting, normally it should also install devDeps

`npm install` only install the `devDependencies` for the working project.

It never install `devDependencies` when you install package as a `dependent`.

**Pull request closed without merging**